pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -796,6 +793,10 @@ func (c *Check) StatementMetrics() (int, error) {
 		}
 		oracleRows = append(oracleRows, oracleRow)
 	}
+
+	c.lastOracleRows = make([]OracleRow, len(oracleRows))
```

Comment:
Sorry I was a bit lost when looking at this PR. To me these 2 lines are the changes for this PR beside the test files update. But `lastOracleRows` was also indicated as `added for tests`. I was not able to connect the code change with what the PR is trying to fix.

Reply:
The primary purpose of this PR i fix the query metrics wrong data. But I also added the test case, which tests surfacing this bug - this was missing. I mentioned this in the PR description>

> New test cases were added.

To make this test, I had to expose the payload slice through the check structure - to test the payload correctness.